clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.h

```diff
@@ -196,6 +196,21 @@ class CIRToLLVMTrapOpLowering : public mlir::OpConversionPattern<cir::TrapOp> {
                   mlir::ConversionPatternRewriter &) const override;
 };
 
+class CIRToLLVMPtrStrideOpLowering
+    : public mlir::OpConversionPattern<cir::PtrStrideOp> {
+  mlir::DataLayout const &dataLayout;
```

Comment:
It's not clear to me why some of these classes store dataLayout and some don't. @bcardosolopes is this an artifact of how the implementation philosophy evolved over time, or is the difference necessary?